### ISSUE TITLE
[MainMenu]: accessibility issues related to the component

### ISSUE THREAD

**Issue by NatalliaAlieva** [epam/UUI#1514]
### Description
Menu items in the opened "More" menu are not accessible from the keyboard

### Precondition
Screen Reader is **OFF** or **ON**

### Steps to Reproduce
1. Open [Main Menu](https://uui.epam.com/documents?category=components&id=mainMenu&mode=doc&skin=UUI4_promo) component.
2. Use `Tab` key to navigate to collapsed "More" button and press `Enter` key to expand it.
3. Use `Tab` key to navigate to the opened items and observe.

### Actual Result
Menu items in the opened "More" menu are not accessible from the keyboard.

https://github.com/epam/UUI/assets/57751283/37d6cc36-a366-48bf-a8d3-975304b095e7

### Expected Result
All functionality including menu items in the opened "More" menu should be accessible from the keyboard.
**Failed WCAG Checkpoint:** 2.1.1 Keyboard (Level A)
https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html

### Description
Part of navigation links are not available when content is zoomed up to 200%

### Precondition
Screen Reader is **ON** or **OFF**

### Steps to Reproduce
1. Open [Main Menu](https://uui.epam.com/documents?category=components&id=mainMenu&mode=doc&skin=UUI4_promo) component.
2. Set the size of the adaptive panel so that the Main Menu (hamburger) button is displayed and click on it.
3. Zoom in the content to 200% and observe.

### Actual Result
"Action items" and "Subscriptions" links are not available when content is zoomed up to 200% and no vertical scroll is provided.

https://github.com/epam/UUI/assets/57751283/6cff37d7-b729-45d2-8128-7e1d2fe9b7f5

### Expected Result
All functionality should be available when content is zoomed up to 200%. 
**Failed WCAG Checkpoint:** 1.4.4 Resize text (Level AA)
https://www.w3.org/WAI/WCAG21/Understanding/resize-text.html

### Description
Incorrect implementation of aria attributes in navigation elements.

### Steps to Reproduce
1. Open [Main Menu](https://uui.epam.com/documents?id=mainMenu&mode=doc&category=components#responsive) component -> Responsive example.
2. Open DevTools and observe the code for the "More" navigation element.
3. Expand the "More" menu and observe the code for nested items.

### Actual Result
The role="menuitem" added to the for nested items (e.g. Companies). The aria-haspopup="true" and role=menuitem" added to the "More".
<img width="1504" alt="Screenshot 2024-12-23 at 14 05 16" src="https://github.com/user-attachments/assets/645b60d8-ab47-499a-90de-973690c82320" />
<img width="1512" alt="Screenshot 2024-12-23 at 14 10 09" src="https://github.com/user-attachments/assets/a3d6d5be-080f-404c-b084-8f9d6b358cba" />

### Expected Result
The role="menuitem" and aria-haspopup="true" attributes should be removed from navigation elements, as they are not menu items.
https://www.w3.org/WAI/ARIA/apg/patterns/landmarks/examples/navigation.html

### Description
"More" dropdown subitems are wrapped in a modal dialog.

### Steps to Reproduce
1. Open [Main Menu](https://uui.epam.com/documents?id=mainMenu&mode=doc&category=components#responsive) component -> Responsive example.
2. Open DevTools and observe the code for the "More" dropdown.

### Actual Result
The "More" dropdown is wrapped in a modal dialog.
<img width="1358" alt="Screenshot 2024-12-23 at 14 17 17" src="https://github.com/user-attachments/assets/18cbd915-c7ad-4639-8163-fa97014b60e3" />

### Expected Result
The “More” button is implemented as a native <button> with: `aria-haspopup=menu` OR `aria-haspopup=true`, `aria-expanded`, and `aria-controls` attributes.
The submenu is rendered with `role="menu"`, and its items have `role="menuitem"`.
The dropdown container **must not have** role="dialog" or aria-modal="true".
https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/
https://www.w3.org/WAI/ARIA/apg/patterns/menubar/

### Description
State and change of state of "More" button is not announced by screen reader.

### Precondition
Screen Reader is **ON**

### Steps to Reproduce
1. Open [Main Menu](https://uui.epam.com/documents?id=mainMenu&mode=doc&category=components) component.
2. Use `Tab` key to navigate to collapsed "More" button and note the announcement.
3. Press Enter key to expand "More" button and note the announcement.

### Actual Result
Current state (collapsed) and change of state (expanded/collapsed) of "More" button is not announced by screen reader.

https://github.com/user-attachments/assets/e8edecb9-ff76-46d0-a4cf-e11f0a68bcdc

### Expected Result
Current state and change of state should be announced by screen reader.

#### Recommendation
To indicate the expanded/collapsed state of the button provide aria-expanded="false" attribute when button is collapsed and aria-expanded="true" when button is expanded.
**Failed WCAG Checkpoint:** 4.1.2 Name, Role, Value (Level A)
https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html

### Description
The navigation links in the site header lack accessible states, including the currently active tab.

### Precondition
Screen Reader is **ON**

### Steps to Reproduce
1. Open uui.epam.com site and select any tab (e.g. "Components") in the header.
2. Use `Tab` and `Shift + Tab` keys to navigate through the header.
3. Observe the output announced by the screen reader for each navigation link, including the active tab.

### Actual Result
The screen reader announces the accessible name and role but does not indicate the state that the link is currently active

https://github.com/user-attachments/assets/0ed55e93-3094-42be-8600-2c08259465eb

<img width="1368" alt="Screenshot 2025-01-07 at 16 43 46" src="https://github.com/user-attachments/assets/44b670a9-8ed3-4a36-ac90-94ea69344cbd" />

### Expected Result
A current link from the list of navigation links should have the appropriate  accessible state
<img width="1491" alt="Screenshot 2025-01-07 at 16 42 57" src="https://github.com/user-attachments/assets/fd353b8a-1089-43ae-9d19-f4233bf8089b" />

#### Recommendation
Set aria-current="page" attribute to current navigation link to indicate the current page within a set of navigation links to screen reader users.
https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA5
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current
**Failed WCAG Checkpoint:** 4.1.2 Name, Role, Value (Level A)
https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html

**Comment by Sardor2** [comment]
@NatalliaAlieva, I've checked the functionality, it seems like it has been fixed.

**Comment by NatalliaAlieva** [comment]
> @NatalliaAlieva, I've checked the functionality, it seems like it has been fixed.

Hello @Sardor2!

You are partially right. The first case is fixed, but the second one is still reproduced.

**Comment by NatalliaAlieva** [comment]
Released in 6.4.0 ver.
